app/hook/context.ts

```diff
@@ -0,0 +1,7 @@
+import { useContext } from "react"
+
+import { ProviderContext } from "~app/context/provider"
+
+export const useProvider = () => {
+  return useContext(ProviderContext)
+}
```

Comment:
I'm wonder why we need to further wrap use `ProviderContext` instead of using `useContext(ProviderContext)` directly.
It seems `useProvider`( or `useProviderContext`) not doing anything else but just import `useContext`.

Reply:
You threw out a good question. Unwrapping it looks good to me. I could do it in another PR. 🫡